### ISSUE TITLE
Cannot get any attribute from vsphere_virtual_machine

### ISSUE THREAD

**Issue by gaeldb** [terraform-providers/terraform-provider-vsphere#558]
Hi there,

When using data.vpshere_virutal_machine on a template vm, I cannot get any attribute from it (id, guest_id, ...).
Strange thing : VM UUID seems to be empty in debug trace.

### Terraform Version && vSphere Provider Version

Terraform v0.11.7
provider.vsphere v1.6.0

### Affected Resource(s)

- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
data "vsphere_datacenter" "datacenter" {
    name = "mydc"
}

data "vsphere_virtual_machine" "template" {
  name          = "templatecentos7"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

output "uuid" {
  value = "${data.vsphere_virtual_machine.template.id}"
}

output "guest_id" {
  value = "${data.vsphere_virtual_machine.template.guest_id}"
}
```

### Debug Output

```
2 error(s) occurred:

* output.guest_id: Resource 'data.vsphere_virtual_machine.template' not found for variable 'data.vsphere_virtual_machine.template.guest_id'
* output.uuid: Resource 'data.vsphere_virtual_machine.template' not found for variable 'data.vsphere_virtual_machine.template.id'
```
Full debug output : https://gist.github.com/gaeldb/344c2325280a65c430971dedf08a0476
where UUID seems to be empty : 
```
[DEBUG] VM search for "templatecentos7" completed successfully (UUID "")
[ERROR] root: eval: *terraform.EvalWriteOutput, err: Resource 'data.vsphere_virtual_machine.template' not found for variable 'data.vsphere_virtual_machine.template.guest_id'
```

Thanks a lot !

**Comment by gaeldb** [comment]
In data_source_vsphere_virtual_machine.go line 105, _virtualmachine.Properties(vm)_ return an empty string Uuid.
Exploring why...

**Comment by gaeldb** [comment]
Ow, govc doesn’t know the UUID neither...

$ govc vm.info templatecentos7
```
Name:           templatecentos7
  Path:         /RD Datacenter/vm/templates/templatecentos7
  UUID:         
  Guest name:   Red Hat Enterprise Linux 7 (64-bit)
  Memory:       4096MB
  CPU:          2 vCPU(s)
  Power state:  poweredOff
  Boot time:    <nil>
  IP address:   
  Host:         xxxxxxxxx 
```

**Comment by gaeldb** [comment]
This is not a bug from Terraform.
The template was created with Packer, which always unset UUID.
according to:
https://github.com/hashicorp/packer/issues/5953
https://groups.google.com/forum/#!topic/packer-tool/1FNRLHpimZg

Just adding a test to catch an empty UUID and inform user.

**Comment by gaeldb** [comment]
commit #559

**Comment by Outek** [comment]
I have almost the same error:


* vsphere_virtual_machine.vm: error sending customization spec: Customization of the guest operating system 'windows9Server64Guest' is not supported in this configuration. Microsoft Vista (TM) and Linux guests with Logical Volume Manager are supported only for recent ESX host and VMware Tools versions. Refer to vCenter documentation for supported configurations.

**Comment by Outek** [comment]
How can i prevent, that packer unset the uuid?

**Comment by Outek** [comment]
I installed the latest VMware Tools (today, it's 10.2.5 from https://packages.vmware.com/tools/releases/index.html) and everything is working fine.

**Comment by bill-rich** [comment]
This was addressed in @gaeldb's PR #559. It was just merged today. Thanks everyone for getting this sorted out!

**Comment by uberjew666** [comment]
> I have almost the same error:
> 
> * vsphere_virtual_machine.vm: error sending customization spec: Customization of the guest operating system 'windows9Server64Guest' is not supported in this configuration. Microsoft Vista (TM) and Linux guests with Logical Volume Manager are supported only for recent ESX host and VMware Tools versions. Refer to vCenter documentation for supported configurations.

@Outek - I had this issue also but it turned out I haven't added the "version" parameter. This needs to be set to 13 which then support windows9Server64Guest. Also the guest_os_type parameter should be "windows9srv-64".
